docs/adrs/0262-custom-label-support.md

```diff
@@ -0,0 +1,43 @@
+# ADR 262: Custom Label Support
+**Date**: 2020-01-14
+
+**Status**: Proposed
+
+## Context
+
+In the case of a single repo having multiple self-hosted runners, actions do not currently support any way of selecting which runner or kind of runner a job should run on, other than by architecture or OS.
+
+Example use cases for this could be:
+
+* Your project involves some machine learning models. You want to add a second runner with a GPU for testing your models, and to have your model tests job use this runner, but other jobs to use the runner without a GPU. 
+* You want to use GitHub actions to deploy your app, and for security you use a self-hosted runner inside your isolated network. If you have a separate isolated network for each environment (e.g. testing, staging, production), you would need to run your deploy job on a different runner depending on which environment you are deploying to.
+* Any other situation where you would like to select which self-hosted runner to use for a job based on the capabilities or characteristics of the runner.
+
+GitHub actions already support selecting a job runner based on labels, and automatically give runners labels based on OS and arch, so a natural extension would be to allow operators of self-hosted runners to add their own labels, so that users can select job runners based on any operator-defined characteristic.
+
```

Comment:
Another scenario for config time labels is someone automating setting up self-hosted runners (VMs, K8s, etc.).  When it's automated, there should be a way to automate applying labels during the automated configuration.  That means that it's more than an interim solution - even if we have the full blown way to add, edit, delete from the web UI (coming) this would still be needed.

Reply:
Good point! In fact, automated deployment of the runner to kubernetes is exactly what I'm doing :)